Automattic/woocommerce-payments
# Prevent WooPay opt-in checks from running when WooPayments is not available

Fixes #9942

#### Changes proposed in this Pull Request
This PR adds checks to prevent initializing WooPayments when a cart does not need a payment and to prevent WooPay opt-in checks from running on those cases.


#### Testing instructions
- Make sure you're not logged in to WooPay nor the merchant site
- Create a 0 dollar digital subscription
- Add to cart
- Go to the blocks checkout page
- Type all the fields but leave the phone number empty. Don't use autofill as it might fill in a phone number
- Make sure the order is placed without any problems or errors
- Test other types of zero dollar products
- Test with other payment methods available and non zero dollar products
- Test trial subscription with a non-zero dollar renewal
- Test other scenarios that might fail


-------------------


**Post merge**
